File: web3-libs/mod5/playground/aula2/balanceOf.js
// web3-libs/mod5/playground/aula2/balanceOf.js
import { http, createPublicClient } from 'viem'; // Importando a biblioteca VIem
import readline from "readline"; // Importando readline para entrada do usuário

// 1. Configure o client com a URL do Anvil
export const publicClient = createPublicClient({
  transport: http('http://127.0.0.1:8545'), // URL do Anvil
});

// 2. Obtenha o endereço da conta via readline input
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

rl.question("Digite o endereço da conta: ", async (address) => {
  try {
    // 3. Usando a função correta para obter o saldo
    const balance = await publicClient.getBalance({ 
      address: address // Passando o endereço como um objeto
    });
    console.log("Saldo em wei:", balance.toString()); // Convertendo para string para exibir
  } catch (error) {
    console.error("Erro ao consultar o saldo:", error);
  }
  rl.close();
});

File: web3-libs/mod3/playground/aula2/balance_of.py
# 1. Instale e importe a biblioteca web3.py
from web3 import Web3

# 2. Configure o provider
RPC_URL = "http://127.0.0.1:8545"
web3 = Web3(Web3.HTTPProvider(RPC_URL))

# 3. Obtenha o endereço da conta via readline input
address = input("Digite o endereço da conta: ")
try:
    balance = web3.eth.get_balance(address)
    print("Saldo em wei:", balance)
except Exception as error:
    print("Erro ao consultar o saldo:", error)
